chain/client-primitives/src/types.rs

```diff
@@ -327,7 +327,8 @@ impl SyncStatus {
         let _span =
             debug_span!(target: "sync", "update_sync_status", old_value = ?self, ?new_value)
                 .entered();
-        *self = new_value
+        crate::metrics::SYNC_STATUS_COUNTER.with_label_values(&[&format!("{new_value}")]).inc();
```

Comment:
The new value may contain some hashes and other arbitrary data. Generally speaking I don't think such data is suitable as lables in metrics. Unless you have a good reason to put the extra info in there I would suggest just setting the label to the SyncStatus enum name.

Reply:
I don't understand your comment.
The label value uses `strum::Display` which converts a enum value to a string like `BodySync`, without extra details. It's `Display`, not `Debug`.